packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -505,11 +505,12 @@ function validateClassAttr(vnode: VBaseElement, elm: Element, renderer: Renderer
     let nodesAreCompatible = true;
     let readableVnodeClassname;
 
-    const elmClassName = getProperty(elm, 'className');
+    const elmClassName = getAttribute(elm, 'class');
+
     if (!isUndefined(className) && String(className) !== elmClassName) {
         // className is used when class is bound to an expr.
         nodesAreCompatible = false;
-        readableVnodeClassname = className;
+        readableVnodeClassname = JSON.stringify(className);
```

Comment:
`data.className` is typed as `string`. Can you add a comment here to say why `JSON.stringify` is used?